src/kwio.sh

```diff
@@ -99,23 +99,76 @@ function success()
   colored_print GREENCOLOR "$@"
 }
 
-# Ask for yes or no
+# Allows the user to select a given question with yes or no answers. Shows a
+# default option (as a capitalized letter). If the user inserts anything
+# different from a "y" or a "yes" (capitalization doesn't matter here) it will
+# be treated as a "no".
 #
 # @message A string with the message to be displayed for the user.
+# @default_option A string (simply y or n) to set the default answer for the
+#                 user. If this parameter is not given, or if it is invalid, it
+#                 will be used 'n'.
 #
-# Returns:
+# Return:
 # Return "1" if the user accept the question, otherwise, return "0"
-#
 # Note: ask_yN return the string '1' and '0', you have to handle it by
 # yourself in the code.
 function ask_yN()
 {
-  local message="$*"
+  local message="$1"
+  local default_option="${2:-'n'}"
+  local yes='y'
+  local no='N'
+
+  if [[ "$default_option" =~ ^([yY][eE][sS]|[yY])+$ ]]; then
+    yes='Y'
+    no='n'
+  fi
+
+  message="$message [$yes/$no]"
+  response=$(ask_with_default "$message" "$default_option" false)
```

Comment:
The `false` option is not needed here.

Reply:
Actually, it is. The default behavior of the ask_with_default is to print a message in this format: `{message} ({defaul_option}): `. The existence of the third string, in this case, false, is what changes this to print just `{message}: `. Since we want in the ask_yN function to print a message like `{message} [y|N]: `, we change this behavior and append to the original message inside ask_yN the possible options.

Just to be sure, I tested it here and go the following results:
When the false parameter is omitted, using this command `ask_yN 'Do you want to do this?' 'y'` prints `Do you want to do this? [Y/n] (y):`, while when there is the false parameter, the exact same command prints this: `Do you want to do this? [Y/n]: ` .

Reply:
I must have missed something then, my bad.

Reply:
No worries, when you said it I was unsure about it too, I thought we may have taken something out that wasn't supposed to.